pydantic/datetime_parse.py

```diff
@@ -243,6 +248,9 @@ def parse_duration(value: StrBytesIntFloat) -> timedelta:
     if kw.get('seconds') and kw.get('microseconds') and kw['seconds'].startswith('-'):
         kw['microseconds'] = '-' + kw['microseconds']
 
+    if kw.get('scientific_notation'):
+        kw['seconds'] = ("%.17s" % kw.pop('scientific_notation')).rstrip('0').rstrip('.')
```

Comment:
```suggestion
        kw['seconds'] = ("%.17s" % kw.pop('scientific_notation')).rstrip('0').rstrip('.')
```
this needs a comment to explain what's going on, also no need to use percent subsitution, you can use an f-string or format.

Also, I'm not quite clear what you're trying to do, but I would guess you can do it with standard string formatting, no need for `rstrip`.

Reply:
I've changed, it's ok for you ?